src/test/unit/PaymentCoordinatorUnit.t.sol

```diff
@@ -0,0 +1,1613 @@
+        reenterer.prepare(targetToUse, msgValueToUse, calldataToUse, bytes("ReentrancyGuard: reentrant call"));
+
+        cheats.expectRevert();
+        paymentCoordinator.payForRange(rangePayments);
+    }
+
+    // Revert with 0 length strats and multipliers
+    function testFuzz_Revert_WhenEmptyStratsAndMultipliers(
+        address avs,
+        bool retroactivePaymentsEnabled,
+        uint256 startTimestamp,
+        uint256 duration,
+        uint256 amount
+    ) public filterFuzzedAddressInputs(avs) {
+        cheats.assume(avs != address(0));
+        cheats.prank(paymentCoordinator.owner());
+        paymentCoordinator.setRetroactivePaymentsEnabled(retroactivePaymentsEnabled);
+
+        // 1. Bound fuzz inputs to valid ranges and amounts
+        IERC20 paymentToken = new ERC20PresetFixedSupply("dog wif hat", "MOCK1", mockTokenInitialSupply, avs);
+        amount = bound(amount, 1, mockTokenInitialSupply);
+        duration = bound(duration, 0, MAX_PAYMENT_DURATION);
+        duration = duration - (duration % calculationIntervalSeconds);
+        startTimestamp;
+        if (retroactivePaymentsEnabled) {
+            startTimestamp = bound(
+                startTimestamp,
+                uint256(_maxTimestamp(GENESIS_PAYMENT_TIMESTAMP, uint64(block.timestamp) - MAX_RETROACTIVE_LENGTH)),
+                block.timestamp + uint256(MAX_FUTURE_LENGTH)
+            );
+        } else {
+            startTimestamp = bound(startTimestamp, block.timestamp, block.timestamp + uint256(MAX_FUTURE_LENGTH));
+        }
+
+        // 2. Create range payment input param
+        IPaymentCoordinator.RangePayment[] memory rangePayments = new IPaymentCoordinator.RangePayment[](1);
+        IPaymentCoordinator.StrategyAndMultiplier[] memory emptyStratsAndMultipliers;
+        rangePayments[0] = IPaymentCoordinator.RangePayment({
+            strategiesAndMultipliers: emptyStratsAndMultipliers,
+            token: paymentToken,
+            amount: amount,
+            startTimestamp: uint64(startTimestamp),
+            duration: uint64(duration)
+        });
+
+        // 3. call payForRange() with expected revert
+        cheats.prank(avs);
+        cheats.expectRevert("PaymentCoordinator._payForRange: no strategies set");
+        paymentCoordinator.payForRange(rangePayments);
+    }
+
+    // Revert with exceeding max duration
+    function testFuzz_Revert_WhenExceedingMaxDuration(
+        address avs,
+        bool retroactivePaymentsEnabled,
+        uint256 startTimestamp,
+        uint256 duration,
+        uint256 amount
+    ) public filterFuzzedAddressInputs(avs) {
+        cheats.assume(avs != address(0));
+        cheats.prank(paymentCoordinator.owner());
+        paymentCoordinator.setRetroactivePaymentsEnabled(retroactivePaymentsEnabled);
+
+        // 1. Bound fuzz inputs to valid ranges and amounts
+        IERC20 paymentToken = new ERC20PresetFixedSupply("dog wif hat", "MOCK1", mockTokenInitialSupply, avs);
```

Comment:
you can reuse the starting block of a lot of these

Reply:
You mean `startTimestamp` for the rangePayment right? Think its better to fuzz the inputs